app/models/user.rb

```diff
@@ -16,8 +16,8 @@ class User < ActiveRecord::Base
   ###################################
 
   has_many :my_video_papers, :foreign_key => 'owner_id', :class_name => 'VideoPaper', :dependent => :destroy
-  has_many :video_papers, -> { distinct }, :through=> :shared_papers
   has_many :shared_papers, :dependent => :destroy
```

Comment:
NOTE: as of rails 5.1 an error is thrown (ActiveRecord::HasManyThroughOrderError) if a `:through` relation is defined before the association it references (`:shared_papers` in this case)